Title: Práctica "El Efecto Fotoeléctrico"  

Author: Javier Martín Cosgaya, Pelayo Campa González, Manuel Alises Núñez, Almudena Ortega Bruna  

Date: 1 de Diciembre de 2020

# EL EFECTO FOTOELÉCTRICO

Cuando la Energía de la luz $E$ incide en un metal, libera una energía mínima $\phi$ para que el electrón se extraiga del metal e intente llegar a la otra placa. Si a este supuesto, le conectamos un cable a ambos metales, se generará una Energía Cinética $E_k$ siempre y cuando $E$ sea mayor que $\phi$ para que el electrón tenga la interacción suficiente para llegar a la otra placa.
$$
E_k = E - \phi
$$

Si además aplicamos una diferencia de potencial $V$ en sentido contrario al movimiento de los electrones, podemos provocar el frenado del movimiento de los mismos. Por lo tanto, dicha Energía Cinética dependerá de la diferencia del potencial por la carga del electrón que se mueve:
$$
e\cdot V = E_k = E - \phi
$$

Sabiendo que la Energía Cinética depende (a su vez) de la masa y velocidad de la partícula:
$$
E_k = \frac{1}{2}\cdot m\cdot v^2
$$
Y la carga del electrón:
$$
e = 1.66 \mathrm{x} 10^{-19} \mathrm{C}
$$




## Primera parte del Ejercicio: demostrar que el primer término de la ecuación tiene unidades de J
Sustituímos por unidades del S.I. aquellos elemenos de la ecuación que conocemos, excepto $V$.

$$
\mathrm{C} \cdot \mathrm{V} = \mathrm{Kg} \cdot \left(\frac{m}{s}\right)^2 
$$


Si las unidades fundamentales del Coulombio $C$:

$$
\mathrm{C} = \mathrm{A} \cdot \mathrm{s}
$$

Podríamos decir que:

$$
\mathrm{A\cdot s}\cdot V = \mathrm{Kg\cdot \big(\frac{m}{s}\big)^2} \rightarrow V = \mathrm{Kg} \cdot \frac{m^2}{A}\cdot \mathrm{s^3}
$$

Apoyándonos en la Segunda Ley de Newton para poder sacar las unidades fundamentales del Newton:
$$
F = m\cdot a \rightarrow \mathrm{N} = \mathrm{Kg\cdot \frac{m}{s^2}} 
$$

Sustituímos en la ecuación anterior para sacar las unidades fundamentales de $V$:
$$
V = \mathrm{\frac{N\cdot m}{A\cdot s}}
$$

Por lo tanto:
$$
e\cdot V = \mathrm{A\cdot s\cdot \frac{N\cdot m}{A\cdot s}} \rightarrow e\cdot V = \mathrm{N\cdot m} = \mathrm{J}
$$

## Ejercicio de Calentamiento
Sabiendo que la recta en dos puntos se define como:
$$
\frac{x - x_1}{x_2 - x_1} =  \frac{y - y_1}{y_2 - y_1}
$$
1. Primero despejamos la $y$:
$$
\frac{x - x_1}{x_2 - x_1}\, \big(y_2 - y_1\big) = y - y_1 \rightarrow \frac{\big(x - x_1\big)\cdot \big(y_2 - y_1\big)}{x_2 - x_1} = y - y_1
$$
$$
y = \frac{\big(x - x_1\big)\cdot \big(y_2 - y_1\big)}{x_2 - x_1} + y_1
$$
2. Deshago los paréntesis calculándo las operaciones:
$$
y = \frac{x\cdot y_2 - x\cdot y_1 - x_1\cdot y_2 + x_1\cdot y_1}{x_2 - x_1} + \frac{y_1\big(x_2 - x_1\big)}{x_2 - x_1}
$$
$$
y = \frac{x\big(y_2 - y_1\big)- x_1\cdot y_2 + x_1\cdot y_1 + x_2\cdot y_1 - x_1\cdot y_1}{x_2 - x_1}
$$
3. Anulamos términos idénticos y damos forma a la ecuación:
$$
y = \frac{x\big(y_2 - y_1\big)}{x_2 - x_1} + \frac{x_2\cdot y_1 - x_1\cdot y_2}{x_2 - x_1}
$$
Conclusión, siguendo el modelo $y = mx + b$:
$$
m = \frac{y_2 - y_1}{x_2 - x_1}\, \rightarrow \, b = \frac{x_2\cdot y_1 - x_1\cdot y_2}{x_2 - x_1}
$$

# SIMULACIÓN EFECTO FOTOELÉCTRICO
Elaboramos una simulación con los colores del arcoiris como referencia de longitudes de onda. Como los electrones no empiezan a moverse hasta un determinado color, solo realizamos la simulación cuando podemos detectar algo de movimiento. Así podremos calcular las constantes con más precisión.

Para la recogida de los datos de muestra se ha utilizado el applet Java del proyecto PhET de la universidad de Colorado. Concretamente en su version HTML, gracias a la tecnología CheerpJ.
CheerpJ es compilador 100% compatible con Java que permite convertir cualquier aplicación Java en una aplicacion WEB. Para ello, las convierte en HTML5 / JavaScript.
Desde su versión 2.0, CheerJ soporta WebAssemply en su entorno de ejecución para una mejora del rendimiento.

In [1]:
var regression = require('./regression.js');

const c = 3e8;
const e = 1.66e-19;
var waveLength = [486, 451, 403].map(x =>x*1e-9) ; //Longitudes de onda para la simulación del metal del Sodio
var voltageMeasured =[-0.20,-0.4,-0.8];
var frecuency = [];
var data = []; 
var layout;
                 
function Plotly(data, layout, $$) {
    $$ = $$ || global.$$;
    $$.mime({ "application/vnd.plotly.v1+json": { data: data, layout: layout } });
}

Para la primera tarea, calcularemos mediante javascript los valores de frecuencia $f$ utilizando el valor de longitud de onda $\lambda$ que hemos escogido para realizar la siguiente simulación.
Aplicando el *Teorema de Fourier* : $\lambda = v / f$ , y sabiendo que la velocidad constante que se produce coincide con la velocidad de la luz ($c = 3\cdot10^{8} \mathrm{m/s}$): 
$$
\lambda\cdot \mathrm{m} = \frac{c\cdot \Big(\mathrm{\frac{m}{s}}\Big)}{f}\, \rightarrow\, f = \frac{c\cdot \Big(\mathrm{\frac{m}{s}}\Big)}{\lambda\cdot \mathrm{m}}\, \rightarrow\, f = \frac{c}{\lambda}\cdot \mathrm{\Big(\frac{1}{s}\Big)}
$$
Siendo las unidades $\mathrm{\frac{1}{s}} = \mathrm{Hz}$

Mediante el siguiente código obtenemos la frecuencia de las longitudes de onda:

In [2]:
for(i=0;i<waveLength.length;i++){
    frecuency[i] = c/waveLength[i];
    console.log(frecuency[i].toExponential()); 
}

6.17283950617284e+14
6.651884700665189e+14
7.444168734491314e+14


Rellenamos la tabla requerida en el ejercicio:

| Color | Longitud de Onda $(\lambda)$ | Frecuencia $(f)$ |
|:-----:|:----------------------------:|:------------------:|
|Rojo|699nm|$4.29\cdot 10^{14}Hz$| 
|Naranja|599nm|$5\cdot 10^{14}Hz$|
|Amarillo|575nm|$5.21\cdot 10^{14}Hz$|
|Verde|533nm|$5.62\cdot 10^{14}Hz$|
|Cyan|486nm|$6.17\cdot 10^{14}Hz$|
|Azul|451nm|$6.65\cdot 10^{14}Hz$|
|Violeta|403nm|$7.44\cdot 10^{14}Hz$|

La longitud de onda $\lambda$ es inversamente proporcional a la frecuencia $f$, esto es, cuando una onda electromagnética con una frecuencia mas alta tiene una longitud de onda más corta, y viceversa.
Realizamos el ajuste de una recta para pintar la gráfica de los valores adquiridos, y así visualizar la tendencia:





Hallada la pendiente y sabiendo que:
$$
e\, V = 2\, \pi\, \hbar\, f - \phi\, \rightarrow\, V = \frac{2\, c\, \pi\, \hbar}{e}\cdot \frac{1}{\lambda} - \frac{\phi}{e}
$$
A partir de la ecuación anterior y sabiendo que la pendiente siguiendo la gráfica tiene las unidades: $m = \mathrm{\frac{V}{m^{-1}}}$ ( ya que es el Voltaje partido de la inversa de la longitud de onda), que $\mathrm{V = \frac{J}{C}}$, que $\mathrm{C = A\cdot s}$ y que la ecuación de la recta es $ y = m\, x + b$ y la carga de un $e = 1.66x10^{-19} \mathrm{C} $, podemos concluir:
$$
\hbar = \frac{m\cdot e}{2\, c\cdot \pi}\cdot \mathrm{\frac{\Big(\frac{V}{m^{-1}}\Big)\cdot C}{\frac{m}{s}}}\, \rightarrow\, \mathrm{\frac{\Big(\frac{J/A\cdot s}{m^{-1}}\Big)\cdot A\cdot s}{\frac{m}{s}}}\, \rightarrow\, J\cdot s
$$
Las unidades de la energía de arranque:
$$
\phi = b\cdot e\cdot \mathrm{\frac{J}{C}\cdot C}\, \rightarrow\, \mathrm{J}
$$


In [3]:
var waveLengthInverse = waveLength.map(lambda => 1/lambda);

var dataToAdjust = voltageMeasured.reduce((tuplesWavelengthVoltage, b, i) => {
  tuplesWavelengthVoltage.push([waveLengthInverse[i],b]);
  return tuplesWavelengthVoltage;
}, []);

console.log(dataToAdjust);
var result = regression.linear(dataToAdjust, {precision: 15});
var m = result.equation[0]; //pendiente
var b = result.equation[1]; //ordenada en el origen
console.log('La pendiente es = ' + m + ' y la ordenada en el origen es b = ' + b);

[
  [ 2057613.1687242799, -0.2 ],
  [ 2217294.9002217297, -0.4 ],
  [ 2481389.5781637714, -0.8 ]
]
La pendiente es = -0.000001425750704 y la ordenada en el origen es b = 2.744265375600105


In [4]:
// Calculamos la constante de Planck y la energía mínima:
var planckConstant = (m * e)/(2 * c * (mathUnits.pi));
console.log('La constante de Planck es: ' + planckConstant);

var minEnergy = b * e;
console.log('La Energía de arranque para el Sodio es: ' + minEnergy);

La constante de Planck es: -1.255597839276202e-34
La Energía de arranque para el Sodio es: 4.555480523496175e-19


En la siguiente línea calculamos los voltajes de frenado mediante la pendiente y la ordenada en el origen:
$$
V_{calculado} = m\cdot \frac{1}{\lambda} + b
$$

In [5]:
var braking_voltage = waveLengthInverse.map(reverse_lambda => reverse_lambda * m + b);
console.log (braking_voltage);

[ -0.1893780482682077, -0.417044389366636, -0.7935775623651553 ]


Con estos valores recogidos, vamos a realizar el ajuste a una recta para realizar el calculo de la constante de Planck y φ.
Para relizarlo, vamos a utilizar el paquete "regression".
Este paquete tambien esta disponible en UNPKG.
UNPKG es un respositorio global, accesible y rápido vía URL, que apoyandose en CDNs mejora el rendimiento de nuestro hosting.
Está licenciado por GPL y es gratuito por lo que su uso es cada vez mas extendido aunque aún esta por debajo de otros servicios de la misma naturaleza como pueden ser jsDelivr o cdnjs.
Los CDNs, ("Content Delivery Network"), son servidores de contenido dispersos geograficamente por la red para agilizar y liberar a nuestro hosting de la descarga de recursos como código JavaScript, CSS.. optimizando el acceso a la Web.

In [6]:
var data =[ {
   x: waveLengthInverse,
   y: voltageMeasured,
   name: 'Longitud de Onda/Voltaje Medido',
   mode: 'lines+markers',
   type: 'scatter'},     
   {
   x: waveLengthInverse,
   y: braking_voltage,
   name: 'Longitud de Onda/Voltaje Calculado',
   mode: 'lines+markers',
   type: 'scatter'} 
   ];

var layout = {
  title: 'Gráfica datos experimentales',
  xaxis: { title: 'Longitud de onda',
    titlefont: { family: 'Courier New, monospace', size: 18, color: '#7f7f7f' }
  },
  yaxis: {
    title: 'Voltaje', titlefont: {
      family: 'Courier New, monospace', size: 18, color: '#7f7f7f' }
  }
};

 Plotly(data,layout);

# SIMULACIÓN CON 4 METALES
Declaramos las variables que contienen los datos medidos con el API de Java del Efecto fotoeléctrico, siendo:
<p>$wavelengthZ$ -> Longitudes de onda del Zinc</p>
<p>$wavelengthC$ -> Longitudes de onda del Cobre</p>
<p>$wavelengthCal$ -> Longitudes de onda del Calcio</p>
<p>$wavelengthS$ -> Longitudes de onda del Sodio</p>

In [1]:
var wavelengthZ = [262,246,222,201].map(x =>x*1e-9); //Trace0 -> Longitudes de onda del Zinc
var wavelengthC = [241,228,207,189].map(x =>x*1e-9); // Trace1 -> Longitudes de onda del Cobre
var wavelengthCal = [378,344,298,261].map(x =>x*1e-9); // Trace2 -> Longitudes de onda del Calcio
var wavelengthS = [466,415,349,299].map(x =>x*1e-9); // Trace3 -> Longitudes de onda del Sodio
var voltagemeasured = [-0.2,-0.6,-1.2,-1.8]; //Voltajes de frenado
var wavelength = [wavelengthZ, wavelengthC, wavelengthCal, wavelengthS]; //Array que contiene las longitudes de onda de todos los metales
var data = []; 
var layout;
var regression = require('./regression.js');


function Plotly(data, layout, $$) {
    $$ = $$ || global.$$;
    $$.mime({ "application/vnd.plotly.v1+json": { data: data, layout: layout } });
}

Esta función calculará los valores de la pendiente y la ordenada para después devolver el valor del Voltaje de Frenado Calculado:

In [2]:
function regressionApplication(data_to_adjust,regression,lambda_1){
  var braking_voltage;
  var result = regression.linear(data_to_adjust,{precision: 15});
  var m = result.equation[0]; //pendiente
  var b = result.equation[1]; //ordenada en el origen
  console.log('La pendiente es = ' + m + ', y la ordenada es = ' + b);
  braking_voltage = lambda_1.map(reverse_lambda => reverse_lambda * m + b);

  return braking_voltage;
}

In [3]:
for(i=0;i<wavelength.length;i++){
  var braking_voltage; 
  var lambda_1 = wavelength[i].map(lambda=> 1/lambda);
  var data_to_adjust = voltagemeasured.reduce((tuple_wl_vm, b ,i) => {
    tuple_wl_vm.push([lambda_1[i],b]);
    return tuple_wl_vm;
  },[]);     
 
 braking_voltage = regressionApplication(data_to_adjust,regression,lambda_1); //Llamada a la función regressionApplication
  
 var traceA = {
   x: lambda_1,
   y: voltagemeasured,
   name: 'Trace'+i,
   mode: 'lines+markers',
   type: 'scatter'};    
  data.push(traceA);
  
   var traceB = {
   x: lambda_1,
   y: braking_voltage,
   name: 'Trace'+i+'s',
   mode: 'lines+markers',
   type: 'scatter'}; 
   data.push(traceB);
}

La pendiente es = -0.000001371677267, y la ordenada es = 5.003581175875551
La pendiente es = -0.0000013863352, y la ordenada es = 5.516306033879675
La pendiente es = -0.00000134183129, y la ordenada es = 3.323599049890657
La pendiente es = -0.000001327004373, y la ordenada es = 2.621424069327308


8

Tabla de función del trabajo $W$ según el material:

| Metal | $\phi/eV$ | Calculado |
|:-----:|:---------:|:---------:|
|Zinc|4.90|5.00| 
|Cobre|5.10|5.52|
|Calcio|2.87|3.32|
|Sodio|2.36|2.62|

A pesar de calcular pendientes muy similares, las ordenadas en el origen tienen un error aproximado de $\pm$ 0.5.

In [4]:
var layout = {
  title: 'Gráfica datos experimentales',
  xaxis: { title: 'Longitud de onda',
    titlefont: { family: 'Courier New, monospace', size: 18, color: '#7f7f7f' }
  },
  yaxis: {
    title: 'Voltaje', titlefont: {
      family: 'Courier New, monospace', size: 18, color: '#7f7f7f' }
  }
};

 Plotly(data,layout);


## Ejercicio Premium

In [5]:
//Medidas obtenidas para una longitud de onda de 100nm y ningún voltaje de frenado. El metal utilizado para las mediciones es el Sodio
var lightIntensity = [0.2,0.4,0.6,0.8,1]; //Intensidad de la luz en %
var current = [0.176,0.353,0.529,0.706,0.882]; //Corriente eléctrica = Nº Electrones
var regression = require('./regression.js');
var data = [];
   
function Plotly(data,layout,$$){
    $$ = $$ || global.$$;
    $$.mime({"application/vnd.plotly.v1+json": {data:data, layout:layout}});

}

In [6]:
var data = [{
  x: current,
  y: lightIntensity,
  name: 'Corriente respecto a Intensidad',
  mode: 'lines+markers',
  type: 'scatter'
}];

var layout = {
  title: 'Gráfica datos experimentales',
  xaxis: { title: 'Intensidad de Corriente',
    titlefont: { family: 'Courier New, monospace', size: 18, color: '#7f7f7f' }
  },
  yaxis: {
    title: 'Intensidad Lumínica', titlefont: {
      family: 'Courier New, monospace', size: 18, color: '#7f7f7f' }
  }
};

 Plotly(data,layout);

## CONCLUSIÓN DE LAS SIMULACIONES
Observamos que cuanta más emisión de radiación, es decir, se aumenta el número de fotones que chocan contra el metal, más electrones se liberan sin modificar su Energía Cinética, y por tanto, sin modificar su velocidad (permanecen ambas constantes).
Cuantos más electrones llegan al otro lado de la placa, más corriente lee el amperímetro.
Podemos concluir, que a más intensidad lumínica mayor es la corriente obtenida, es decir que la Intensidad de Corriente es proporcional a la Intensidad Lumínica o Nº de fotones emitidos.

# ALGO DE HISTORIA
La historia del descubrimiento del efecto foto eléctrico es toda una aventura, para la que hay que viajar a finales del siglo XIX y principios del XX.

En 1864, James Clerk Maxwell consiguió uno de los mayores logros de la física del siglo XIX, describiendo un fenómeno que denominó electromagnetismo.

![](https://lh3.googleusercontent.com/proxy/yVZUjoL2E8oeYJ8ssQ1-3aIcFXD1cQYvrE2NvUFPYIrwwHWZ27Su2XO2JzgD50QKHoP1f_j2frv-AlK6_F1dopbhrAAWHD0loOSQbKyz9_sRP8FodG0Mkf_qzyP0OID6Xj53KDo)

Dos décadas después, fue H. Hertz quien aportó la corroboración experimental de la existencia de estas ondas electromagnéticas, para entonces, desgraciadamente Maxwell ya había fallecido. O quizá, afortunadamente, porque a su vez, Hertz, accidentalmente, descubrió el efecto fotoeléctrico que tiraba por tierra la teoría clásica del electromagnetismo de Maxwell, que incluso Hertz daba por sentada.

Pero no fue capaz de desarrollar ninguna teoría que diera explicación a este fenómeno. A su favor, es justo decir que el hecho de que aún no se conociera la existencia del electrón no le ayudo mucho.

En 1900, tres años después de descubrir la existencia del electrón, surge en escena Max Planck, quien a la desesperada había recurrido a solución absurda e ilógica para dar explicación a este efecto. Esto es, que la luz son cuantos de energía, es decir, unos paquetes discretos cuya energía crece con la frecuencia de la luz y de una constante muy pequeña. Que más tarde recibirá su nombre, la constante de Planck.

La física clásica no ve con buenos ojos esta teoría “alocada”, e incluso “suicida” podríamos decir, de Planck, y sigue intentando dar explicación a porque se produce este efecto fotoeléctrico. Fue en 1902, el ayudante de Hertz, Philipp Lenard quien continuo la investigación, llegando a descubrir que la energía de estos electrones no dependía de la intensidad de la luz sino únicamente de su longitud de onda, pero sin llegar a dar la explicación de efecto en sí.

No fue hasta 1905, cuando Albert Einstein, es su cuarto artículo, curiosamente el menos conocido y por el contrario por el que consiguió el premio Nobel de Física en 1921, quien consiguiera demostra el efecto fotoeléctrico.

Para ello, Einstein, se basó en la hipótesis cuántica de Planck, sobre los cuantos de luz. Su aplicación al efecto fotoeléctrico, le permitió explicar un fenómeno físico, la dualidad onda-partícula, porque la luz tiene ambas naturalezas, lo que conocemos por fotón.

Para ello, Einstein, al igual que hizo Planck se vio obligado a recurrir a conceptos nuevos para explicar el misterio, conceptos que supusieron el origen de una nueva rama de la física, la cuántica.

Este hecho produjo una gran rivalidad entre Einstein y Lenard, ya que sintió que le había robado su descubrimiento. Su rencor creció cuando Einstein recibió el Nobel de Física.

La rivalidad fue tal que Einstein llego a publicar a siguiente cita:
“…admiro a Lenard como maestro de la física experimental, [pero] todavía no ha producido nada destacado en física.”

Esta rivalidad, fue generalizada entre dos vertientes de la física de la época, la física “judía” y la física “aria”. La primera, a la que pertenecen Planck y Eisntein por su ascendencia judía, era una física más aplicada frente a la física más teórica del otro bando. El hecho de que en esa época fuera aumentando el antisemitismo, tampoco ayudo. Por ello, Einstein comenzó a recibir críticas e insultos con tintes raciales y un ataque de la prensa, tanto la genérica como la especializada.

Un dato curioso, es que tanto Planck y Einstein, quienes fueron los que dieron los primeros pasos a lo que hoy conocemos física cuántica, no tardaron posteriormente a negar cualquier implicación que tuvieran con ella. 
